R/kwic.R

```diff
@@ -135,39 +135,84 @@ is.kwic <- function(x) {
     inherits(x, "kwic")
 }
 
+#' @rdname kwic
+#' @method as.data.frame kwic
+#' @export
+as.data.frame.kwic <- function(x, ..., window = NULL, separator = NULL) {
+    
+    if (is.null(window))
+        window <- attr(x, "window")
+    else
+        window <- check_integer(window, 1, 1, 0)
+    
+    if (is.null(separator))
+        separator <- attr(x, "separator")
+    else
+        separator <- check_character(separator)
+    
+    attrs <- attributes(x)
+    x$pre <- rep("", nrow(x))
+    x$keyword <- rep("", nrow(x))
+    x$post <- rep("", nrow(x))
+    if (nrow(x)) {
+        toks <- attrs$tokens[x$docname]
+        lis_pre <- as.list(tokens_select(toks, startpos = pmax(x$from - window, 1), endpos = x$from - 1))
```

Comment:
I discovered a bug when processing phrases that our tests did not detect, because they only compared parts of the result to the expectation. Fixed in a4741b1.

```r
> toks <- tokens(c(doc1 = "a a a b c d d d", doc2 = "a b c d e"))
> kwic(toks, phrase("b c"), window = 2)
Keyword-in-context with 2 matches.                                
 [doc1, 4:5] a a b | b c | c d d
 [doc2, 2:3]   a b | b c | c d e
```